modules/internal/NetworkAtomics.chpl

```diff
@@ -21,986 +21,263 @@ pragma "atomic module"
-      chpl_comm_atomic_cmpxchg_int64(te, td, this.locale.id:int(32), this._v, ret);
-      return ret;
-    }
-    inline proc compareExchangeWeak(expected:int(64), desired:int(64),
-                                    order:memory_order = memory_order_seq_cst):bool {
-      return this.compareExchange(expected, desired);
-    }
-    inline proc compareExchangeStrong(expected:int(64), desired:int(64),
-                                      order:memory_order = memory_order_seq_cst):bool {
-      return this.compareExchange(expected, desired);
+    inline proc _localeid: int(32) {
+      return this.locale.id:int(32);
     }
 
-    inline proc fetchAdd(value:int(64), order:memory_order = memory_order_seq_cst):int(64) {
-      var v = value;
-      var ret:int(64);
-      chpl_comm_atomic_fetch_add_int64(v, this.locale.id:int(32), this._v, ret);
-      return ret;
-    }
-    inline proc add(value:int(64), order:memory_order = memory_order_seq_cst) {
-      var v = value;
-      chpl_comm_atomic_add_int64(v, this.locale.id:int(32), this._v);
-    }
+    inline proc const read(order:memory_order = memory_order_seq_cst): bool {
+      pragma "insert line file info" extern externFunc("get", int(64))
+        proc atomic_get(ref result:int(64), l:int(32), const ref obj:int(64)): void;
 
-    inline proc fetchSub(value:int(64), order:memory_order = memory_order_seq_cst):int(64) {
-      var v = value;
-      var ret:int(64);
-      chpl_comm_atomic_fetch_sub_int64(v, this.locale.id:int(32), this._v, ret);
-      return ret;
-    }
-    inline proc sub(value:int(64), order:memory_order = memory_order_seq_cst) {
-      var v = value;
-      chpl_comm_atomic_sub_int64(v, this.locale.id:int(32), this._v);
+      var ret: int(64);
+      atomic_get(ret, _localeid, _v);
+      return ret:bool;
     }
 
-    inline proc fetchOr(value:int(64), order:memory_order = memory_order_seq_cst):int(64) {
-      var v = value;
-      var ret:int(64);
-      chpl_comm_atomic_fetch_or_int64(v, this.locale.id:int(32), this._v, ret);
-      return ret;
-    }
-    inline proc or(value:int(64), order:memory_order = memory_order_seq_cst) {
-      var v = value;
-      chpl_comm_atomic_or_int64(v, this.locale.id:int(32), this._v);
-    }
+    inline proc write(value:bool, order:memory_order = memory_order_seq_cst): void {
+      pragma "insert line file info" extern externFunc("put", int(64))
+        proc atomic_put(ref desired:int(64), l:int(32), ref obj:int(64)): void;
 
-    inline proc fetchAnd(value:int(64), order:memory_order = memory_order_seq_cst):int(64) {
-      var v = value;
-      var ret:int(64);
-      chpl_comm_atomic_fetch_and_int64(v, this.locale.id:int(32), this._v, ret);
-      return ret;
-    }
-    inline proc and(value:int(64), order:memory_order = memory_order_seq_cst) {
-      var v = value;
-      chpl_comm_atomic_and_int64(v, this.locale.id:int(32), this._v);
+      var v = value:int(64);
+      atomic_put(v, _localeid, _v);
```

Comment:
I share Greg's concerns about 'v' vs. '_v' and '_localeid'.

Reply:
Hmm, thinking about it more, I'm less concerned about ``_localeid``. I wouldn't object to making it use parentheses.

Reply:
The only reason for the temporary is because the runtime functions take `v` by ref and our default intent for bool/int/uint/real is `const in`. I'm not sure why the runtime takes it by ref though. 

I'm ok with making _loacleid a parenful function or going back to `this.locale.id:int(32)` -- whatever you guys prefer.

I went from `this._v` to `_v` to match the processor atomics -- I can go back to `this._v` or rename `v` to something else, or investigate the runtime functions more to see if there's any reason `v` is taken by ref.

Reply:
I don't see any reason why the runtime functions take the operand(s) by ref.  We obviously only have the comm=ugni implementations of these, and those definitely don't need the addresses. All they do with those is copy the pointed-to operands into the the FMA Post descriptors.

Reply:
In the interest of progress, we could retain the `v` variable usage for now and make a separate, lower priority issue to change the comm layer AMO operands to by-value and get rid of the `v` variables.  `_localeid()` might be nice to do now, though.